app/services/customers/generate_checkout_url_service.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Customers
+  class GenerateCheckoutUrlService < BaseService
+    def initialize(customer:)
+      @customer = customer
+      @provider_customer = customer&.provider_customer
+
+      super
+    end
+
+    def call
+      return result.not_found_failure!(resource: 'customer') if customer.blank?
+      return result.service_failure!(
+        code: 400,
+        message: 'no payment provider linked to this customer'
```

Comment:
Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)